jest/consistent-test-it does not handle describe.each properly

#### Description

When configuring the rule `jest/consistent-test-it`, it doesn't work for `describe.each` calls.

####  To Reproduce

```
$ cat .eslintrc.js 
module.exports = {
  plugins: ["jest"],
  rules: {
    "jest/consistent-test-it": ["error", { fn: "test", withinDescribe: "it" }],
  },
};

$ cat package.json 
{
  "name": "jest-consistent-test-it-bug",
  "scripts": {
    "test": "eslint test.js"
  },
  "devDependencies": {
    "eslint": "^7.22.0",
    "eslint-plugin-jest": "^24.3.1"
  }
}

$ cat test.js 
var tests = [["jest/consistent-test-it", false]];

test("is valid", function () {});

describe("is valid", function () {
  it("is valid", function () {});
})

describe.each(tests)("%s", function (name, bool) {
  test("is valid, but should not be", function () {});

  it("is not valid, but should be", function () {
    expect(bool).toBe(true);
  });
});

$ npm run test

> test
> eslint test.js


/home/kelly/wsp/jest-bug/test.js
  12:3  error  Prefer using 'test' instead of 'it'  jest/consistent-test-it

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

npm ERR! code 1
npm ERR! path /home/kelly/wsp/jest-bug
npm ERR! command failed
npm ERR! command sh -c eslint test.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kelly/.npm/_logs/2021-03-16T19_23_17_839Z-debug.log
```

--- a/src/rules/__tests__/consistent-test-it.test.ts
+++ b/src/rules/__tests__/consistent-test-it.test.ts
@@ -1,4 +1,5 @@
 import { TSESLint } from '@typescript-eslint/experimental-utils';
+import dedent from 'dedent';
 import resolveFrom from 'resolve-from';
 import rule from '../consistent-test-it';
 import { TestCaseName } from '../utils';
@@ -168,6 +169,46 @@ ruleTester.run('consistent-test-it with fn=test', rule, {
           data: {
             testKeywordWithinDescribe: TestCaseName.test,
             oppositeTestKeyword: TestCaseName.it,
+          },
+        },
+      ],
+    },
+    {
+      code: 'describe.each``("foo", () => { test.each``("bar") })',
+      output: 'describe.each``("foo", () => { it.each``("bar") })',
+      options: [{ fn: TestCaseName.it }],
+      errors: [
+        {
+          messageId: 'consistentMethodWithinDescribe',
+          data: {
+            testKeywordWithinDescribe: TestCaseName.it,
+            oppositeTestKeyword: TestCaseName.test,
+          },
+        },
+      ],
+    },
+    {
+      code: dedent`
+        describe.each()("%s", () => {
+          test("is valid, but should not be", () => {});
+
+          it("is not valid, but should be", () => {});
+        });
+      `,
+      output: dedent`
+        describe.each()("%s", () => {
+          it("is valid, but should not be", () => {});
+
+          it("is not valid, but should be", () => {});
+        });
+      `,
+      options: [{ fn: TestCaseName.test, withinDescribe: TestCaseName.it }],
+      errors: [
+        {
+          messageId: 'consistentMethodWithinDescribe',
+          data: {
+            testKeywordWithinDescribe: TestCaseName.it,
+            oppositeTestKeyword: TestCaseName.test,
           },
         },
       ],

--- a/src/rules/consistent-test-it.ts
+++ b/src/rules/consistent-test-it.ts
@@ -8,6 +8,7 @@ import {
   createRule,
   getNodeName,
   isDescribe,
+  isEachCall,
   isTestCase,
 } from './utils';
 
@@ -120,7 +121,7 @@ export default createRule<
         }
       },
       'CallExpression:exit'(node) {
-        if (isDescribe(node)) {
+        if (isDescribe(node) && !isEachCall(node)) {
           describeNestingLevel--;
         }
       },

